### ISSUE TITLE
macOS (High Sierra, 10.13.2)

### ISSUE THREAD

**Issue by lame-impala-life** [kmatheussen/radium#1079]
### 1. Recipe
* Step 1: Open JackPilot
* Step 2: Configure with README suggestions (with Hog Mode)
* Step 3: Click "Start" (JACK runs)
* Step 4: Move "Radium" and "Kill Jack" applications into "Applications" through the sidebar of Finder
* Step 5: Double-click "Radium"
* Step 6: Get the Security Message (Radium from an unknown source...)
* Step 7: OK the warning in System Preferences > Security & Privacy
* Step 8: Double-click "Radium" (Splash image will show)

I expected this to happen:
Radium to stay open, and hopefully show the tracker interface.

This is what actually happened:
After the splash, a white panel appears with messages printing quickly (alike starting up Linux with boot-terminal messages enabled) and after a split-second the panel disappears. The Radium icon in the dock accordingly shrinks and peters out as well. Radium appeared to crash.

* I've tried this recipe 3 times.

* The unexpected behavior has happened 3 times / approximately 100% of the times / exactly 100% of the times.

### 2. Which version of Radium are you using? Do you know if an earlier version has worked?
Radium 5.x.x listed on the download page, retrieved 2017 12 08 (December 8th, 2017)

### 3. If relevant, which operating system did you run Radium on?
macOS High Sierra 10.13.2

**Comment by kmatheussen** [comment]
Hi, don't know what happened here.
Does the crashreporter show up?
Do you have ATI or Nvidia graphics?
Suggestions: 1. Try to NOT use hog mode for jack. 2. Try to start jack  like this instead, in a terminal:
```
jackd -ddummy
```
 (you won't hear anything, but if this works we know that it's got something to do with jack)

Can you try to run radium from the console and post the output? That is likely to narrow down the problem a lot. Try to run this command:
```
Radium.app/Contents/MacOS/Radium 2>log.log
```
And afterwards post the file log.log

**Comment by kmatheussen** [comment]
Do you have retina display by the way? Any information about the graphics card and display is useful. Crashes like these are usually related to graphics.

**Comment by kmatheussen** [comment]
I hope most of the high sierra / sierra problems will go away when I upgrade the build system. Especially Qt (the main graphics library used by Radium) is very old. I've been waiting a while for osxcross (https://github.com/tpoechtrager/osxcross) to be upgraded to at least macOS 10.12. I think it's currently at 10.10. However, Radium is currently compiled against 10.8. It's very hard to upgrade Qt until I upgrade osxcross, but perhaps I should just upgrade to 10.10 first, and later upgrade to 10.12 when it's available. Since you are a subscriber, I'm inclined to do this.

**Comment by lame-impala-life** [comment]
* CrashReporter does not show up.

* This MacBook uses Intel Integrated Graphics (see below). If I recall correctly, the chip is quite notorious.

* It does use a Retina display.

### System Information
**From "About This Mac"**
```
MacBook (Retina, 12-inch, Early 2016)
Processor 1.1 GHz Intel Core m3
Memory 8 GB 1867 MHz LPDDR3
Graphics Intel HD Graphics 515 1536 MB
Serial Number <Not necessary?>
```
```
Built-in Display
12-inch (2304 x 1440)
Intel HD Graphics 515 1536 MB
```
****

**Comment by lame-impala-life** [comment]
Disabling "Hog Mode" in JackPilot did not change the outcome overall. However, the interface (I believe) did appear for a split-second once out of ~6 runs. CrashReporter did open once as well.

**Comment by lame-impala-life** [comment]
<img width="1280" alt="screen shot 2017-12-09 at 10 41 56 am" src="https://user-images.githubusercontent.com/30834126/33797464-f760cc2e-dccd-11e7-8021-832869dba7e7.png">

**Comment by lame-impala-life** [comment]
Attempting to run the jackd bit in terminal happens upon a funky output.
```
jackdmp 1.9.11
Copyright 2001-2005 Paul Davis and others.
Copyright 2004-2016 Grame.
jackdmp comes with ABSOLUTELY NO WARRANTY
This is free software, and you are welcome to redistribute it
under certain conditions; see the file COPYING for details
Unknown driver "dummy"
```
* I take jackdmp would not be the intended program (though invoked by 'jackd'). I will need to find the correct one again.
* The lack of a 'dummy' driver is even more off-putting.

**Comment by kmatheussen** [comment]
It would be very interesting to see the content of the crash reporter. By crash reporter, I meant the Radium crashreporter (made for radium). But seeing the content of the macOS crash reporter would be very useful as well.

Can you look under Library/Logs/DiagnosticReports in your home area and see if there are any files in there that contains the "Radium"? The last Radium-file is likely to be very useful to look at.

**Comment by kmatheussen** [comment]
> The lack of a 'dummy' driver is even more off-putting.

Yes. Very strange. There is a dummy driver for jack, of course. I don't know why it's not available on OSX.

**Comment by lame-impala-life** [comment]
I do not see the Radium file in the DiagnosticReports directly. I have not been able to coax the Mac CrashReporter into firing either.

**Comment by lame-impala-life** [comment]
[log.log](https://github.com/kmatheussen/radium/files/1544901/log.log)
[stdoutstuffmacos.pdf](https://github.com/kmatheussen/radium/files/1544902/stdoutstuffmacos.pdf)

**Comment by kmatheussen** [comment]
Thank you. There wasn't enough information to track down what's happening, but it helped a little bit. The last thing that was printed to the terminal happened right before trying to resize the OpenGL window, so something with OpenGL is my best bet.

Perhaps you can try to delete the config file? It's the file ".radium/config" in your home area.
```
rm ~/.radium/config
```

If that didn't help, maybe adding "opengl_draw_in_separate_process = false" to the config file helps:

```
echo "opengl_draw_in_separate_process = false" >>~/.radium/config
```

**Comment by lame-impala-life** [comment]
Neither command did the trick.

**Comment by kmatheussen** [comment]
V5.4.0 is compiled with the latest SDK, Clang, and Qt. Maybe it helped, but probably not as I've read about others having problem with OpenGL applications crashing when running on High Sierra.

**Comment by kmatheussen** [comment]
Forgot this. The crashing should be fixed in 5.4.2.

**Comment by kmatheussen** [comment]
I assume this problem was fixed in 5.4.1 since I haven't noticed any crashes myself or gotten crashreports from others since then.
